java/sage/Carny.java

```diff
@@ -1159,61 +1617,87 @@ private void stdProcessing()
     newWPMap.keySet().removeAll(watchedPotsToClear);
 
     if (Sage.DBG) System.out.println("CARNY Negative Energy Size: " + (blackBalled.size() + stoners.length));
-    for (Airing badAir : blackBalled)
+    for (int i = 0, blackBalledSize = blackBalled.size(); i < blackBalledSize; i++)
```

Comment:
I learned a little about this while I was working on the new Scheduler. I changed everything that made sense into an iterator with the assumption that JIT would know best and optimize things. I was very disappointed.